.github/workflows/unit_tests.yaml

```diff
@@ -1,107 +1,29 @@
-      run: vermin --backport argparse --violations --backport typing -t=2.7- -t=3.6- -vvv var/spack/repos
-  # Run style checks on the files that have been changed
-  style:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # @v2
-      with:
-        fetch-depth: 0
-    - uses: actions/setup-python@b55428b1882923874294fa556849718a1d7f2ca5 # @v2
-      with:
-        python-version: '3.10'
-    - name: Install Python packages
-      run: |
-        pip install --upgrade pip six setuptools types-six
-    - name: Setup git configuration
-      run: |
-        # Need this for the git tests to succeed.
-        git --version
-        . .github/workflows/setup_git.sh
-    - name: Run style tests
-      run: |
-          share/spack/qa/run-style-tests
-  # Check which files have been updated by the PR
-  changes:
-    runs-on: ubuntu-latest
-    # Set job outputs to values from filter step
-    outputs:
-      core: ${{ steps.filter.outputs.core }}
-      packages: ${{ steps.filter.outputs.packages }}
-      with_coverage: ${{ steps.coverage.outputs.with_coverage }}
-    steps:
-    - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # @v2
-      if: ${{ github.event_name == 'push' }}
-      with:
-        fetch-depth: 0
-    # For pull requests it's not necessary to checkout the code
-    - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
-      id: filter
-      with:
-        # See https://github.com/dorny/paths-filter/issues/56 for the syntax used below
-        filters: |
-          core:
-          - './!(var/**)/**'
-          packages:
-          - 'var/**'
-    # Some links for easier reference:
-    #
-    # "github" context: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context
-    # job outputs: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs
-    # setting environment variables from earlier steps: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
-    #
-    - id: coverage
-      # Run the subsequent jobs with coverage if core has been modified,
-      # regardless of whether this is a pull request or a push to a branch
-      run: |
-        echo Core changes: ${{ steps.filter.outputs.core }}
-        echo Event name: ${{ github.event_name }}
-        if [ "${{ steps.filter.outputs.core }}" == "true" ]
-        then
-          echo "::set-output name=with_coverage::true"
-        else
-          echo "::set-output name=with_coverage::false"
-        fi
-
   # Run unit tests with different configurations on linux
-  unittests:
-    needs: [ validate, style, changes ]
+  unit-tests-ubuntu:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['2.7', '3.6', '3.7', '3.8', '3.9', '3.10']
+        python-version: ['2.7', '3.6', '3.10']
```

Comment:
This will cut ~1h30 cpu-time for each CI workflow run on a core PR and use three less runners per PR. 

I have to note though that there is at least a counterexample to this strategy being the same as testing on the whole range. Using Python versions `[3.4 - 3.6]` we would catch bugs using 3.5 that we wouldn't using only the other two (because of dicts). I think the decision here is ultimately on @tgamblin since it's a trade-off between the time spent and the tests we perform.

Maybe, would it be easy to test on these versions for the PRs and test on every version on push to `develop`? If that is possible we might reap most of the time savings _and_ be notified rapidly if something breaks the compatibility with a specific Python version.

Reply:
I could make it so develop would test the full range while PRs test the smaller set, or even restore this. Even without it the response time on PR or push should be rather better since we don't slam CI with everything up front. Still looking at how to collapse down the bootstrap test, it's not too bad but needs a bit of scripting.

Reply:
We should still run the full range somewhere and make sure we fix issues that come up.  There are some tests, particularly the ones for `spack.util.unparse`  and `spack.util.package_hash`, that are AST-sensitive -- those can vary in tricky ways with some of the small changes between versions.   The ones for spec hashing are also sensitive to dict ordering.

If that only happens on `develop` I don't think it's so bad.  I think it's the kind of thing that should happen on enqueue once we have merge queue, but we can watch commits to `develop` and make sure to fix them in the meantime.  Thoughts?